### PR TITLE
Bootstrap - Enhanced app structure and some work on chorus-dt dashboard

### PR DESCRIPTION
**Pull request by Tiyop**
The goal of this PR is to:

1. Enhance the structure of the app folder
- apps/ for web pages. should only contain layout and callbacks
- components/ for page components (figures, cards..)
- assets/ static images, css and js scripts
- utils/ mainly for data generation and wrangling files
2. Develop a first layout for travels (chorus dt) related dashboard

### CODE REVIEW

File: app/src/components/html_components.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import dash_core_components as dcc
+import dash_html_components as html
+import dash_bootstrap_components as dbc
+import plotly.graph_objects as go
+
+def build_figure_container(title, id, footer, fig=go.Figure()):
```
Comment (antoan2) [inline app/src/components/html_components.py:7]: Nice one.

File: app/src/index.py [hunk 2]
```diff
@@ -1,21 +1,37 @@
...
+
+app.layout = dbc.Container(
     [
         html.Div(id="div-url-redirect"),
         dcc.Location(id="url", refresh=False),
-        html.H1("Outil d'aide à la création de BEGES", style={"text-align": "center", "margin-top": "5%"}),
-        html.Hr(),
+        navbar,
+        #html.H1("Outil d'aide à la création de BEGES", style={"text-align": "center", "margin-top": "5%"}),
+        #html.Hr(),
```
Comment (antoan2) [inline app/src/index.py:30]: You should delete these comments.

File: app/src/apps/chorus_dt.py [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+import dash_bootstrap_components as dbc
+import plotly.graph_objects as go
+from dash.dependencies import Output, Input, State
+
+from app import app
+from utils.chorus_dt_handler import ch
+from components.html_components import build_figure_container
+from components.figures_templates import xaxis_format
+
+def make_donut_by_prestation():
```
Comment (antoan2) [inline app/src/apps/chorus_dt.py]: I like the `Generic Crossfilter Recipe` in the [dash callback tutorial page](https://dash.plotly.com/interactive-graphing). So I just propose we use functions named `get_...` with arguments used to get the right data.
```suggestion
def get_donut_by_prestation_type(code_structure=None):
```

File: app/src/apps/chorus_dt.py [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+    # Load chorus dt data based on chosen code_structure
+    # TODO: improve and standardize data import logic
+    chorus_dt_df = ch.get_structure_data(code_structure=None)
+    prestation_df = chorus_dt_df.groupby(['prestation_type'])['cumulative_distance'].sum().reset_index()
+    fig = go.Figure(
+        data=[go.Pie(labels=prestation_df.prestation_type, values=prestation_df['cumulative_distance'], hole=.3)])
+    fig.update_layout(plot_bgcolor='white', template='plotly_white', margin={'t': 30, 'r': 30, 'l': 30})
+    return fig
+
+def make_emissions_timeseries():
```
Comment (antoan2) [inline app/src/apps/chorus_dt.py]:
```suggestion
def get_emissions_timeseries(code_structure):
```

File: app/src/apps/chorus_dt.py [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+)
+def update_emissions_timeseries(selected_entity):
+    return make_emissions_timeseries()
+
+
+@app.callback(
+    Output("donut-by-prestation", "figure"), [Input("selected-entity", "children")]
+)
+def update_donut_by_prestation(selected_entity):
+    return make_donut_by_prestation()
```
Comment (antoan2) [inline app/src/apps/chorus_dt.py]:
```suggestion
    return get_donut_by_prestation(selected_entity)
```

File: app/src/apps/chorus_dt.py [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+def display_graphs(selected_entity):
+    organisation, service = selected_entity.split(";")
+    return organisation + " / " + service
+
+
+@app.callback(
+    Output("timeseries-chorus-dt", "figure"), [Input("selected-entity", "children")]
+)
+def update_emissions_timeseries(selected_entity):
+    return make_emissions_timeseries()
```
Comment (antoan2) [inline app/src/apps/chorus_dt.py:152]:
```suggestion
    return get_emissions_timeseries(selected_entity)
```

File: app/src/apps/chorus_dt.py [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+    Output("selected-entity-show", "children"), [Input("selected-entity", "children")]
+)
+def on_selected_entity_fill_tabs_data(selected_entity):
+    if selected_entity is not None:
+        organisation, service = selected_entity.split(";")
+        return "Organisation : " + organisation + ", Service : " + service
+    else:
+        return "empty"
+
+@app.callback(Output("values-selected'", "children"), [Input("selected-entity", "children")])
```
Comment (antoan2) [inline app/src/apps/chorus_dt.py:144]: I don't really understand the need for this callback.

File: app/src/apps/chorus_dt.py [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+        ], width=8),
+        dbc.Col([
+            build_figure_container(title='Répartition des émissions par type de déplacement', id='donut-by-prestation', \
+                                   footer='Explications..'),
+        ], width=4),
+
+    ])
+], id="div-data-chorus-dt")
+
+@app.callback(
```
Comment (antoan2) [inline app/src/apps/chorus_dt.py]: Maybe we should have some helper somewhere to manage the "selected-entity / children" value and return what we want :
- organization id for a specific dataset
- services id for a specific dataset (we might need to store in selected entities several ids for some organizations (for example for AC))

This function would split the selected-entity using the delimiter ';' and then return the rights ids. Something like this (in an horrible not functionning pseudo code):
``` python
class EntityHandler:
    def get_organization_service_ids(self, selected_entity, dataset):
         orga, services = selected_entity.split(';')
         if dataset == 'chorus-dt':
             orga = self.get_chorus_code(orga)
             services =  [self.get_chorus_code(service) for service in services]
         return organization_code, service_codes
```

**Review by antoan2 — APPROVED** [review]
Good work.
I approve these changes as we need to first push this dashboard and merge everything into master to let other contributors start working on their part of the app.

File: app/src/apps/chorus_dt.py (hunk 7 above)
Comment (Tiyop) [reply app/src/apps/chorus_dt.py:144]: Haha me neither

File: app/src/apps/chorus_dt.py (hunk 8 above)
Comment (Tiyop) [reply app/src/apps/chorus_dt.py]: True - I think this one might require some more in-depth discussion to allow users to:
- Switch between orgs/services in a same view (via filter panel)
- Select multiple orgs/services

Besides, as you are pointing out there should be mapping between chorus/osfi/odrive for orgs and services.

Adding it as an issue.

**Pull request merged**